### PR TITLE
Add update-status itest

### PR DESCRIPTION
**Pull request by sed-i**
This PR adds an "resilience" itest for when update-status fires very often.
(Initially this was prompted by #65, but this itest does not test `--force` behavior anymore.)

### CODE REVIEW

**Comment by rbarry82** [comment]
Should probably be part of #65 in general so we don't end up with more weird out-of-sync patches.

To be clear, I don't thin that this is one, but I'm not sure why it's not just part of the other PR.

As a general repeat of the other comment, though, this seems to me to lack the urgency to push it in (`--force` is not a common workflow) when the problem should be handled from OF, and we a little time to wait.

**Review by balbirthomas — APPROVED** [review]

**Comment by balbirthomas** [comment]
Not a review comment but a topical observation:

This is an excellent test to compare behavior with issue we are having with Loki charm in order to gauge the source of the problem. It is also interesting to observer that the integration tests are failing although in a way different from the failure (hang) we are seeing in the Loki tests. I am not sure if it is the small differences in the test or the bigger differences in the charm that is causing this difference in outcomes.

File: lib/charms/alertmanager_k8s/v0/alertmanager_dispatch.py [hunk 1]
```diff
@@ -279,7 +279,12 @@ def update_relation_data(self, event: RelationEvent = None):
             # a single consumer charm's unit may be related to multiple providers
             if self.name in self.charm.model.relations:
                 for relation in self.charm.model.relations[self.name]:
+                    # Sometimes (e.g. when an app is removed with `--force`), there is a dangling
```
[the remark this answers is not in the document]
Comment (sed-i) [reply lib/charms/alertmanager_k8s/v0/alertmanager_dispatch.py:282]: "`--force` bugs are not bugs", however: https://github.com/canonical/operator/issues/745#issuecomment-1111411892

**Comment by sed-i** [comment]
I removed the `--force` portion of the test. Mind giving this another review @rbarry82?

**Review by rbarry82 — APPROVED** [review]

**Pull request merged**